DEVDOCS-5657 [update]: Tax Provider API, update development flow (#130)
bigcommerce/docs: docs/api-docs/provider-apis/tax/overview.mdx

The [Tax Provider API](/docs/rest-contracts/tax) works in conjunction with a BigCommerce app, so you will need to [build an app](/docs/integrations/apps) that integrates the [Tax Provider API](/docs/rest-contracts/tax).

> You can publish your BigCommerce app so that it is discoverable by anyone or publish their app as unlisted so it can only be installed using a URL.

Reviewer: fixed ✔️

> You can publish your BigCommerce app so that it is discoverable by anyone or publish their app as unlisted so it can only be installed using a URL.

Reviewer: fixed ✔️

> You can publish your BigCommerce app so that it is discoverable by anyone or publish their app as unlisted so it can only be installed using a URL.

Reviewer: fixed ✔️

> You can publish your BigCommerce app so that it is discoverable by anyone or publish their app as unlisted so it can only be installed using a URL.

Reviewer: fixed ✔️

The [Tax Provider API](/docs/rest-contracts/tax) works in conjunction with a BigCommerce app, so you will need to [build an app](/docs/integrations/apps) that integrates the [Tax Provider API](/docs/rest-contracts/tax).

> You can publish your BigCommerce app so that it's discoverable by anyone or publish your app as unlisted so it can only be installed using a URL.

Reviewer: fixed in suggestion

> You can publish your BigCommerce app so that it's discoverable by anyone or publish their app as unlisted so it can only be installed using a URL.

Reviewer 2: seems to be mixed context here , your vs their

Author: Good call, thanks!

> You can publish your BigCommerce app so that it is discoverable by anyone or publish their app as unlisted so it can only be installed using a URL.

Reviewer: fixed ✔️

> You can publish your BigCommerce app so that it is discoverable by anyone or publish their app as unlisted so it can only be installed using a URL.

Reviewer: fixed ✔️

> Tax providers can publish their BigCommerce app so that it's discoverable by anyone or publish their app as unlisted so it can only be installed using a URL.

Author: Removed nuances around "multi-tenant" and private instances.

We're looking to simplify our on-going maintenance here and have adopted an approach that speaks to simplified options from an app listing perspective (either listed or unlisted - end of story).

Reviewer: 👍

Once your tax provider configuration is ready, we'll let you know through email. The email will also include your `provider_id`, which is required when [establishing a connection](#establishing-a-connection) with the [Tax Provider API](/docs/rest-contracts/tax).

| Tax Provider Details | Required / Optional | Value(s) | Description | Example |
| :--- | :--- | :--- | :--- | :--- |
| App ID | Required | Integer | Tells us which tax provider configuration to use after the app is installed. | `123456` |
| Tax provider name | Required | String | Displayed in the control panel (e.g. **Settings > Setup > Tax > Add tax service**). | `Sample Tax` |
| Transaction type | Required | Production, Test | Whether your tax provider will be handling production traffic or conducting test transactions. | `Production` |

> | Partner support email | Required | Email | Used by BigCommerce to contact tax providers, in the case we need to forward a merchant support request or discuss integration improvements. | `support@sampletax.example.com` |

Reviewer: fixed ✔️

> | Partner support email | Required | Email | Used by BigCommerce to contact tax providers, in the case we need to forward a merchant support request or discuss integration improvements. | `support@sampletax.example.com` |

Reviewer: fixed ✔️

> | Partner support email | Required | Email | Used by BigCommerce to contact tax providers, in the case we need to forward a merchant support request or discuss integration improvements. | `support@sampletax.example.com` |

Reviewer: fixed ✔️

**Coverage** | | | | |
| Platform availability | Required | All stores, Limited availability | Whether your tax provider will be available to all stores or a limited number of stores. | `Limited availability` |

> | Supported / unsupported countries | Required | ISO 3166-1 alpha-2 | Comma separated ISO 3166-1 alpha-2 country codes for supported countries. You may also select to support to all countries, or all countries except a list of unsupported countries. | `US,CA,GB,FR,AU,NZ` |

Author: I did action this one.

Once your tax provider configuration is ready, we'll let you know through email. The email will also include your `provider_id`, which is required when [establishing a connection](#establishing-a-connection) with the [Tax Provider API](/docs/rest-contracts/tax).

| Tax Provider Details | Required / Optional | Value(s) | Description | Example |
| :--- | :--- | :--- | :--- | :--- |
| App ID | Required | Integer | Tells us which tax provider configuration to use after the app is installed. | `123456` |
| Tax provider name | Required | String | Displayed in the control panel (e.g. **Settings > Setup > Tax > Add tax service**). | `Sample Tax` |

> | Tax provider type | Required | Production, Sandbox | Hierarchy of tax provider configurations, Production is primary and Sandboxes are secondary, see [sandbox tax provider configuration](#sandbox-tax-provider-configuration) for more information. | `Production` |

Author: We often get confused tax partners wondering what we mean by "Sandbox" and hence we've replaced this concept with Transaction type below - which is more inline with what we care about and uses more common, clearer language.

Reviewer: 👍

| Partner support email | Required | Email | Used by BigCommerce to contact tax provider, in the case we need to forward a merchant support request. | `support@sampletax.example.com` |

> | Links displayed in the control panel | Optional | URL, Title, Description | Link(s) displayed to merchants when they navigate to **Settings > Setup > Tax > _TaxProviderName_**. One or multiple links supported. | `support.sampletax.com`, `Sample Tax Support`, `The Sample Tax Support website.` |

Author: We don't offer this feature any longer (links) and, indeed, have removed it from our UI entirely. We expect apps to offer their own interfaces (and support links), as is warranted.

Reviewer: 👍

> | Partner support email | Required | Email | Used by BigCommerce to contact tax provider, in the case we need to forward a merchant support request or discuss integration improvements. | `support@sampletax.example.com` |

Author: We had one tax partner recently confused around why we needed an email here, given it was only for their own usage - hence we're clarifying that it's not just to communicate between a tax partner and a merchant but also BigCommerce and the tax partner.

Reviewer 2: `Used by BigCommerce to contact tax provider`. reads a little odd.
perhaps `Used by BigCommerce to contact tax providers` or `Used by BigCommerce to contact a tax provider`

Author: Agreed, thanks!

> | Partner support email | Required | Email | Used by BigCommerce to contact tax providers, in the case we need to forward a merchant support request or discuss integration improvements. | `support@sampletax.example.com` |

Reviewer: ```suggestion
| Partner support email | Required | Email | Contact email that BigCommerce uses to forward merchant support requests or discuss integration improvements with you. | `support@sampletax.example.com` |
```

| Transaction type | Required | Production, Test | Whether your tax provider will be handling production traffic or conducting test transactions. | `Production` |
| Partner support email | Required | Email | Used by BigCommerce to contact tax provider, in the case we need to forward a merchant support request or discuss integration improvements. | `support@sampletax.example.com` |
**Coverage** | | | | |

> | Tax provider visibility | Required | Show, Hide | Tells us if tax provider should be displayed to merchants in the control panel before the tax providers app is installed on the associated store. | `Show` |

Author: This isn't something we allow flexibility on anymore (to simplify our maintenance) and speaks to the Platform availability option. Previously we'd often get confused tax partners on this option - no longer!

Reviewer: 👍

| Platform availability | Required | All stores, Private instance | Tells us if tax provider should work on all stores or only on the stores where a store hash has been provided. | `All stores` |

> | Supported store(s) | Required if platform availability is **private instance** | Store hashes | As a private instance, tax provider configuration will only work on store hashes provided. | `dwvjntfqv,epq54yymgq` |

Author: This isn't a feature we offer any more. This is handled by an unlisted app with a limited availability tax provider. The result is that the tax provider manages who has access directly (e.g. if someone gets a hold of the direct link it's up to them to authenticate the account, which is nothing to do with BigCommerce).

In particular this means our maintenance burden is significantly reduced since we're not constantly updating store hashes.

Reviewer: 👍

> | Platform availability | Required | All stores, Limited availability | Whether your tax provider will be available to all stores or a limited number of stores. | `Limited availability` |

Author: It's relatively rare for us to engage with a tax provider that is seeking to offer availability to all stores - hence we've flipped around the language a little here, and also changed the language away from specific store hashes (which we are removing support for).

Reviewer: 👍

> | Supported / unsupported countries | Required | ISO 3166-1 alpha-2 | Comma separated ISO 3166-1 alpha-2 country codes for supported countries. You may also select to support to all countries, or all countries except a list of unsupported countries. | `US,CA,GB,FR,AU,NZ` |

Author: I've attempted to clarify this a little more from the perspective of "all countries" and "unsupported countries", not just focusing on the supported countries.

Reviewer: 👍

  You must liaise with the merchant to provide them with customized URLs. BigCommerce merely consumes the URL values the merchant enters and uses the provided string value for Tax Provider API requests.
</Callout>

> ### Test tax provider (optional)

Author: We've had feedback that the terminology we were using was often confused - so I've adjusted this to focus on the reason tax providers may want to implement a "test tax provider".

I'm not 100% sold on the language in this area in particular.

One specific call out is the language "tax provider" that we use throughout this documentation - sometimes this means a more _technical_ tax provider ("tax provider instance") and sometimes it means the team building the solution ("tax provider"). For larger organisations we'd call the later "tax partner". We have also used the terminology "tax service" to describe the "tax provider instance" in some places.

I'm not completely sold on any particular change in this area, but thought I'd highlight if someone else has some ideas or preferences.

Reviewer: @theromulans I'm thinking of referring to tax provider organizations/partners with "you" (like what I did in the [Tax Profile section](https://github.com/bigcommerce/docs/pull/130/files#diff-6454eb7541f7642d69afa823f3e2b7f3812936c2afd4fdc54094178546abda3fR45)). This would make sense since tax providers organizations/partners/team are the people using the doc (and second-person voice is also a style choice that we use elsewhere in BC docs).

Then we could use "Tax Provider instance" to refer to the technical tax provider service, like what you mentioned.

Reviewer: ^I made suggestions in a couple of places to illustrate what I mean, but I can go through the rest of the doc to make this change

Author: Great, thanks for the guidance here @bc-andreadao!

I've made some further adjustments to try and remove the language confusion here and use the language you've suggested - in particular with regard to perspective: more "You..." and less "The tax provider" third person.

> Tax providers may choose to build a tax provider instance that supports test transactions. This test tax provider instance will sit alongside a primary production tax provider instance, servicing merchant-facing use cases, or to support the tax provider's own development processes.

Reviewer: ```suggestion
Tax providers may choose to build a tax provider instance that supports test transactions. This test tax provider instance will sit alongside a primary production tax provider instance, servicing merchant-facing use cases or supporting the tax provider's own development processes.
```

Tax providers that support test transactions have lower expectations than those supporting production traffic (e.g. timeouts, error handling), and so it is important to differentiate between a production tax provider and a test tax provider when sharing provider details with BigCommerce.

<Callout type="info">

>   We recommend tax providers request to create only one test tax provider. Any development procedures related to testing the tax provider's different estimate, commit, adjust and void Tax Provider API endpoints should be performed externally and decoupled from the BigCommerce platform to avoid unnecessary dependencies.

Reviewer: ```suggestion
  When you develop your app, it is best to test [Tax Provider API](/docs/rest-contracts/tax) endpoints outside of the BigCommerce platform. This helps avoid unnecessary dependencies.
```

  You must liaise with the merchant to provide them with customized URLs. BigCommerce merely consumes the URL values the merchant enters and uses the provided string value for Tax Provider API requests.
</Callout>

### Test tax provider (optional)

> You can build a tax provider instance that supports test transactions. This test tax provider instance will sit alongside a primary production tax provider instance, servicing merchant-facing use cases or supporting the tax provider's own development processes.

Reviewer: ```suggestion
You can build a tax provider instance alongside a primary production tax provider instance. The test tax provider instance can support test transactions, service merchant-facing use cases, or support your own development processes. 
```

Author: Great, I've just adjusted this slightly to specify `a test tax provider instance` in the first sentence.

I've also modified the language in the following paragraph to better match this language ("test tax provider instance").

> Tax provider instances that support test transactions have lower expectations than those supporting production traffic (e.g. timeouts, error handling), and so it's important to differentiate between a production tax provider and a test tax provider when sharing provider details with BigCommerce.

Reviewer: ```suggestion
When you [share provider details](#sharing-provider-details-with-bigcommerce), it's important to specify the transaction type for your tax provider instance. Tax provider instances that support test transactions have lower expectations than those supporting production traffic like timeouts or error handling.
```

Author: Great, I had also considered adding a link back to the sharing provider details section - cheers!

<Callout type="info">

>   We recommend only creating one test tax provider instance. When you develop your app, it's best to test your [Tax Provider API](/docs/rest-contracts/tax) endpoints outside the BigCommerce platform. This helps avoid unnecessary dependencies.

Reviewer: doesn't apply

</Callout>

## Building the app

> A BigCommerce [single-click app](/docs/integrations/apps/guide/types#single-click) must be built to utilize the Tax Provider API to provide tax estimates and submit tax documents. A BigCommerce single-click app provides many benefits. For example, it enables you to promote your solution in the BigCommerce App Marketplace, ask for merchant authorization of API scopes during app install, and allow merchants to configure tax provider settings through an iFrame in the control panel.

Reviewer: ```suggestion
You must build a BigCommerce [single-click app](/docs/integrations/apps/guide/types#single-click) to use the Tax Provider API for tax estimates and document submissions. Using a BigCommerce single-click app offers many benefits. For example, you can promote your solution in the BigCommerce App Marketplace and ask merchants to authorize API scopes during app install. Single-click apps also allow merchants to configure tax provider settings through an iFrame in the control panel.
```

Author: Great reword, cheers.

  You must liaise with the merchant to provide them with customized URLs. BigCommerce merely consumes the URL values the merchant enters and uses the provided string value for Tax Provider API requests.
</Callout>

### Test tax provider (optional)

> You can build a tax provider instance that supports test transactions. This test tax provider instance will sit alongside a primary production tax provider instance, servicing merchant-facing use cases or supporting the tax provider's own development processes.

Reviewer: suggested a fix ✔️

> Tax provider instances that support test transactions have lower expectations than those supporting production traffic (e.g. timeouts, error handling), and so it is important to differentiate between a production tax provider and a test tax provider when sharing provider details with BigCommerce.

Reviewer: fixed ✔️

> Tax provider instances that support test transactions have lower expectations than those supporting production traffic (e.g. timeouts, error handling), and so it is important to differentiate between a production tax provider and a test tax provider when sharing provider details with BigCommerce.

Reviewer: fixed in a suggestion ✔️

<Callout type="info">

>   We recommend only creating one test tax provider instance. When you develop your app, it is best to test your [Tax Provider API](/docs/rest-contracts/tax) endpoints outside of the BigCommerce platform. This helps avoid unnecessary dependencies.

Author: Actioned.

</Callout>

## Building the app

> A BigCommerce [single-click app](/docs/integrations/apps/guide/types#single-click) must be built to utilize the Tax Provider API to provide tax estimates and submit tax documents. A BigCommerce single-click app provides many benefits. For example, it enables you to promote your solution in the BigCommerce App Marketplace, ask for merchant authorization of API scopes during app install, and allow merchants to configure tax provider settings through an iFrame in the control panel.

Author: Look, this sentence sat poorly with me also - but I'll leave it for the professionals 😉

> A BigCommerce [single-click app](/docs/integrations/apps/guide/types#single-click) must be built to utilize the Tax Provider API to provide tax estimates and submit tax documents. A BigCommerce single-click app provides many benefits. For example, it enables you to promote your solution in the BigCommerce App Marketplace, ask for merchant authorization of API scopes during app install, and allow merchants to configure tax provider settings through an iFrame in the control panel.

Reviewer: fixed in a suggestion ✔️

> A BigCommerce [single-click app](/docs/integrations/apps/guide/types#single-click) must be built to utilize the Tax Provider API to provide tax estimates and submit tax documents. A BigCommerce single-click app provides many benefits. For example, it enables you to promote your solution in the BigCommerce App Marketplace, ask for merchant authorization of API scopes during app install, and allow merchants to configure tax provider settings through an iFrame in the control panel.

Reviewer: fixed in a suggestion ✔️

> A BigCommerce [single-click app](/docs/integrations/apps/guide/types#single-click) must be built to utilize the Tax Provider API to provide tax estimates and submit tax documents. A BigCommerce single-click app provides many benefits. For example, it enables you to promote your solution in the BigCommerce App Marketplace, ask for merchant authorization of API scopes during app install, and allow merchants to configure tax provider settings through an iFrame in the control panel.

Reviewer: fixed in a suggestion ✔️

> A BigCommerce [single-click app](/docs/integrations/apps/guide/types#single-click) must be built to utilize the Tax Provider API to provide tax estimates and submit tax documents. A BigCommerce single-click app provides many benefits. For example, it enables you to promote your solution in the BigCommerce App Marketplace, ask for merchant authorization of API scopes during app install, and allow merchants to configure tax provider settings through an iFrame in the control panel.

Reviewer: fixed in a suggestion ✔️

> A BigCommerce [single-click app](/docs/integrations/apps/guide/types#single-click) must be built to utilize the Tax Provider API to provide tax estimates and submit tax documents. A BigCommerce single-click app provides many benefits. For example, it enables you to promote your solution in the BigCommerce App Marketplace, ask for merchant authorization of API scopes during app install, and allow merchants to configure tax provider settings through an iFrame in the control panel.

Reviewer: fixed in a suggestion ✔️

> A BigCommerce [single-click app](/docs/integrations/apps/guide/types#single-click) must be built to utilize the Tax Provider API to provide tax estimates and submit tax documents. A BigCommerce single-click app provides many benefits. For example, it enables you to promote your solution in the BigCommerce App Marketplace, ask for merchant authorization of API scopes during app install, and allow merchants to configure tax provider settings through an iFrame in the control panel.

Reviewer: lol, fixed in a suggestion ✔️

> A BigCommerce [single-click app](/docs/integrations/apps/guide/types#single-click) must be built to utilize the Tax Provider API to provide tax estimates and submit tax documents. A BigCommerce single-click app provides many benefits. For example, it enables you to promote your solution in the BigCommerce App Marketplace, ask for merchant authorization of API scopes during app install, and allow merchants to configure tax provider settings through an iFrame in the control panel.

Reviewer: fixed in a suggestion ✔️

Review our [introduction to building apps](/docs/integrations/apps) guide and use the sidebar to explore topics including: types of apps, managing apps in the dev portal, implementing OAuth, and designing the UI.

Make sure to also review our [app development best practices](/docs/integrations/apps/guide/best-practices) for some tips.

<Callout type="info">

>   If you have registered your app in the Developer Portal but have not submitted it for approval because it is still in development, the app will be in a Draft state. This means your app can only be installed on stores owned by the same email address as the Developer Portal account email.

Reviewer: fixed in suggestion ✔️

>   If you have registered your app in the Developer Portal but have not submitted it for approval because it is still in development, the app will be in a Draft state. This means your app can only be installed on stores owned by the same email address as the Developer Portal account email.

Reviewer: skip, doesn't seem to apply here

</Callout>

## Installing the app

> When the merchant clicks **Install** on your single-click app, it is essential that the app successfully handle the OAuth flow before moving on to the next step of establishing a connection with the [Tax Provider API](/docs/rest-contracts/tax). The [Apps Guide](/docs/integrations/apps/guide/auth) provides an overview of the OAuth flow.

Reviewer: fixed in suggestion ✔️

> When the merchant clicks **Install** on your single-click app, it is essential that the app successfully handle the OAuth flow before moving on to the next step of establishing a connection with the [Tax Provider API](/docs/rest-contracts/tax). The [Apps Guide](/docs/integrations/apps/guide/auth) provides an overview of the OAuth flow.

Reviewer: fixed in suggestion ✔️

> When handling the OAuth flow, you must ensure your app is requesting read and write permissions on the **Information and Settings** API scopes. Additionally, your should store the **client_id** and **access_token** that are received. Both are required later when [establishing a connection with the Tax Provider API](#establishing-a-connection).

Reviewer: fixed in a suggestion ✔️

> When handling the OAuth flow, you must ensure your app is requesting read and write permissions on the **Information and Settings** API scopes. Additionally, your should store the **client_id** and **access_token** that are received. Both are required later when [establishing a connection with the Tax Provider API](#establishing-a-connection).

Reviewer: fixed in a suggestion ✔️

> When handling the OAuth flow, you must ensure your app is requesting read and write permissions on the **Information and Settings** API scopes. Additionally, your should store the **client_id** and **access_token** that are received. Both are required later when [establishing a connection with the Tax Provider API](#establishing-a-connection).

Reviewer: fixed in a suggestion ✔️

## Establishing a connection

> You will be ready to establish a connection with the Tax Provider API when all of the following are true:

Reviewer: fixed in a suggestion ✔️

> You will be ready to establish a connection with the Tax Provider API when all of the following are true:

Reviewer: fixed in a suggestion ✔️

> You will be ready to establish a connection with the Tax Provider API when all of the following are true:

Reviewer: fixed in a suggestion

> * You have shared your tax provider details with BigCommerce through email, and we've replied confirming the tax provider configuration is now ready on all supplied merchant and/or partner test stores (see [Sharing provider details with BigCommerce](#sharing-provider-details-with-bigcommerce)).

Reviewer: fixed in a suggestion

* BigCommerce has shared the `provider_id` of the tax provider configuration with you through email (see [Sharing provider details with BigCommerce](#sharing-provider-details-with-bigcommerce)).
* You have built a draft app that successfully handles the OAuth flow during app installation (see [Building the app](#building-the-app)).

> * During the OAuth flow the app requests that the merchant authorize read and write permissions on the Information and Settings API scopes (see [Installing the app](#installing-the-app)).

Reviewer: doesn't apply

> * During the OAuth flow the `client_id` and `access_token` was received and stored by you (see [Installing the app](#installing-the-app)).

Reviewer: fixed in suggestion

> For context, the [Tax Provider API Connection endpoints](/docs/rest-contracts/tax-app-connection) provide an added layer of security for tax provider connectivity. These basic credentials are used to authenticate each API request to the tax provider from the associated store, allowing you to reject any other unauthorized requests.

Reviewer: fixed in suggestion

> For context, the [Tax Provider API Connection endpoints](/docs/rest-contracts/tax-app-connection) provide an added layer of security for tax provider connectivity. These basic credentials are used to authenticate each API request to the tax provider from the associated store, allowing you to reject any other unauthorized requests.

Reviewer: fixed in suggestion

> An account login flow allows you to capture basic authentication credentials from merchants. It is best to provide a login flow within your app iFrame so that you can choose your own authentication method. You can also provide an account registration flow in your iFrame to help merchants create a new tax provider account. Learn more about [designing the app UI](/docs/integrations/apps/guide/ui).

Reviewer: fixed in suggestion

> An account login flow allows you to capture basic authentication credentials from merchants. It is best to provide a login flow within your app iFrame so that you can choose your own authentication method. You can also provide an account registration flow in your iFrame to help merchants create a new tax provider account. Learn more about [designing the app UI](/docs/integrations/apps/guide/ui).

Reviewer: not applicable

> An account login flow allows you to capture basic authentication credentials from merchants. It is best to provide a login flow within your app iFrame so that you can choose your own authentication method. You can also provide an account registration flow in your iFrame to help merchants create a new tax provider account. Learn more about [designing the app UI](/docs/integrations/apps/guide/ui).

Reviewer: not applicable (not verb form)

> An account login flow allows you to capture basic authentication credentials from merchants. It is best to provide a login flow within your app iFrame so that you can choose your own authentication method. You can also provide an account registration flow in your iFrame to help merchants create a new tax provider account. Learn more about [designing the app UI](/docs/integrations/apps/guide/ui).

Reviewer: not applicable

> In either case, the [Update Connection](/docs/rest-contracts/tax-app-connection#update-a-connection) endpoint should be called after your app has been successfully installed and the merchant's account credentials have been authenticated. You will need to include `store_hash`, `provider_id`, and `X-Auth-Token`(`access_token`) values.

Reviewer: revised

> In either case, the [Update Connection](/docs/rest-contracts/tax-app-connection#update-a-connection) endpoint should be called after your app has been successfully installed and the merchant's account credentials have been authenticated. You will need to include `store_hash`, `provider_id`, and `X-Auth-Token`(`access_token`) values.

Reviewer: fixed in suggestion

> In either case, the [Update Connection](/docs/rest-contracts/tax-app-connection#update-a-connection) endpoint should be called after your app has been successfully installed and the merchant's account credentials have been authenticated. You will need to include `store_hash`, `provider_id`, and `X-Auth-Token`(`access_token`) values.

Reviewer: fixed in suggestion

> In either case, the [Update Connection](/docs/rest-contracts/tax-app-connection#update-a-connection) endpoint should be called after your app has been successfully installed and the merchant's account credentials have been authenticated. You will need to include `store_hash`, `provider_id`, and `X-Auth-Token`(`access_token`) values.

Reviewer: fixed in suggestion

> In either case, the [Update Connection](/docs/rest-contracts/tax-app-connection#update-a-connection) endpoint should be called after your app has been successfully installed and the merchant's account credentials have been authenticated. You will need to include `store_hash`, `provider_id`, and `X-Auth-Token`(`access_token`) values.

Reviewer: fixed in suggestion

> In either case, the [Update Connection](/docs/rest-contracts/tax-app-connection#update-a-connection) endpoint should be called after your app has been successfully installed and the merchant's account credentials have been authenticated. You will need to include `store_hash`, `provider_id`, and `X-Auth-Token`(`access_token`) values.

Reviewer: fixed in suggestion

We recommend calling the [Update Connection](/docs/rest-contracts/tax-app-connection#update-a-connection) endpoint immediately after the app has been successfully installed, otherwise your tax provider will not be displayed when merchants navigate to the **Settings > Setup > Tax** page in the control panel.

The [Get a Connection](/docs/rest-contracts/tax-app-connection#get-connection-status) request can be used at any time to retrieve the connection status of the specified tax provider in the context of a store.

## Configure tax provider settings in the control panel

> Once you have successfully installed your app and provided basic authentication credentials through the [Update Connection](/docs/rest-contracts/tax-app-connection#update-a-connection) request, merchants will be able to adjust their tax provider settings to activate supported functions on their store. 

Author: Actioned.

> Once you have successfully installed your app and provided basic authentication credentials through the [Update Connection](/docs/rest-contracts/tax-app-connection#update-a-connection) request, merchants will be able to adjust their tax provider settings to activate supported functions on their store. 

Reviewer: suggested a fix

> Once you have successfully installed your app and provided basic authentication credentials through the [Update Connection](/docs/rest-contracts/tax-app-connection#update-a-connection) request, merchants will be able to adjust their tax provider settings to activate supported functions on their store. 

Reviewer: already fixed

Review our [introduction to building apps](/docs/integrations/apps) guide and use the sidebar to explore topics including: types of apps, managing apps in the dev portal, implementing OAuth, and designing the UI.

Make sure to also review our [app development best practices](/docs/integrations/apps/guide/best-practices) for some tips.

<Callout type="info">
  If you have registered your app in the Developer Portal but have not submitted it for approval because it is still in development, the app will be in a Draft state. This means your app can only be installed on stores owned by the same email address as the Developer Portal account email.
</Callout>

## Installing the app

When the merchant clicks **Install** on your single-click app, it is essential that the app successfully handle the OAuth flow before moving on to the next step of establishing a connection with the [Tax Provider API](/docs/rest-contracts/tax). The [Apps Guide](/docs/integrations/apps/guide/auth) provides an overview of the OAuth flow.

> When handling the OAuth flow, you must ensure your app is requesting read and write permissions on the **Information and Settings** API scopes. Additionally, your should store the **client_id** and **access_token** that are received. Both are required later when [establishing a connection with the Tax Provider API](#establishing-a-connection).

Reviewer: fixed in a suggestion ✔️

* BigCommerce has shared the `provider_id` of the tax provider configuration with you through email (see [Sharing provider details with BigCommerce](#sharing-provider-details-with-bigcommerce)).
* You have built a draft app that successfully handles the OAuth flow during app installation (see [Building the app](#building-the-app)).
* During the OAuth flow the app requests that the merchant authorize read and write permissions on the Information and Settings API scopes (see [Installing the app](#installing-the-app)).
* During the OAuth flow the `client_id` and `access_token` was received and stored by you (see [Installing the app](#installing-the-app)).

For context, the [Tax Provider API Connection endpoints](/docs/rest-contracts/tax-app-connection) provide an added layer of security for tax provider connectivity. These basic credentials are used to authenticate each API request to the tax provider from the associated store, allowing you to reject any other unauthorized requests.

> An account login flow allows you to capture basic authentication credentials from merchants. It's best to provide a login flow within your app iFrame so that you can choose your own authentication method. You can also provide an account registration flow in your iFrame to help merchants create a new tax provider account. Learn more about [designing the app UI](/docs/integrations/apps/guide/ui).

Reviewer: not applicable

We recommend calling the [Update Connection](/docs/rest-contracts/tax-app-connection#update-a-connection) endpoint immediately after the app has been successfully installed, otherwise your tax provider will not be displayed when merchants navigate to the **Settings > Setup > Tax** page in the control panel.

The [Get a Connection](/docs/rest-contracts/tax-app-connection#get-connection-status) request can be used at any time to retrieve the connection status of the specified tax provider in the context of a store.

## Configure tax provider settings in the control panel

Once you have successfully installed your app and provided basic authentication credentials through the [Update Connection](/docs/rest-contracts/tax-app-connection#update-a-connection) request, merchants will be able to adjust their tax provider settings to activate supported functions on their store. 

Merchants can adjust these settings by navigating to **Settings > Setup > Tax > _TaxProviderName_** in the control panel. 

> Before you can provide tax estimates, the merchant must choose which regions they want you to service. The control panel displays a list of countries and subdivisions that you support. 

Reviewer: not applicable

> If document submission is supported, the merchant can choose to select the **Submit Order Data** checkbox. This will enable document submission on their store, for supported transactions.

Reviewer: fixed in suggestion

>   If you have registered your app in the Developer Portal but have not submitted it for approval because it's still in development, the app will be in a Draft state. This means your app can only be installed on stores owned by the same email address as the Developer Portal account email.

Reviewer: ```suggestion
  When you develop your app, you can register your app in the Developer Portal without submitting it. The app will be in a Draft state. This means your app can only be installed on stores owned by the same email address as that on the Developer Portal account.
```

</Callout>

## Installing the app

> When the merchant clicks **Install** on your single-click app, it's essential that the app successfully handle the OAuth flow before moving on to the next step of establishing a connection with the [Tax Provider API](/docs/rest-contracts/tax). The [Apps Guide](/docs/integrations/apps/guide/auth) provides an overview of the OAuth flow.

Reviewer: ```suggestion
When the merchant clicks **Install** on your single-click app, the app must handle the OAuth flow before moving on to the next step of establishing a connection with the [Tax Provider API](/docs/rest-contracts/tax). The [Apps Guide](/docs/integrations/apps/guide/auth) provides an overview of the OAuth flow.
```

When handling the OAuth flow, you must ensure your app is requesting read and write permissions on the **Information and Settings** API scopes. Additionally, your should store the **client_id** and **access_token** that are received. Both are required later when [establishing a connection with the Tax Provider API](#establishing-a-connection).

## Establishing a connection

> You will be ready to establish a connection with the Tax Provider API when all of the following are true:

Reviewer: ```suggestion
You will be ready to set up a connection with the Tax Provider API when all of the following are true:
```

> * You have shared your tax provider details with BigCommerce through email, and we've replied confirming the tax provider configuration is now ready on all supplied merchant and/or partner test stores (see [Sharing provider details with BigCommerce](#sharing-provider-details-with-bigcommerce)).

Reviewer: ```suggestion
* You have [shared your tax provider details](#sharing-provider-details-with-bigcommerce) with BigCommerce through email, and we've replied confirming the tax provider configuration is now ready on all supplied merchant and/or partner test stores.
```

Author: Great, much prefer to avoid (see XXX) and have this inline 💯

> * BigCommerce has shared the `provider_id` of the tax provider configuration with you through email (see [Sharing provider details with BigCommerce](#sharing-provider-details-with-bigcommerce)).

Reviewer: ```suggestion
* BigCommerce has shared the `provider_id` of the tax provider configuration with you through email after you [shared your tax provider details](#sharing-provider-details-with-bigcommerce).
```

* You have built a draft app that successfully handles the OAuth flow during app installation (see [Building the app](#building-the-app)).
* During the OAuth flow the app requests that the merchant authorize read and write permissions on the Information and Settings API scopes (see [Installing the app](#installing-the-app)).

> * During the OAuth flow the `client_id` and `access_token` was received and stored by you (see [Installing the app](#installing-the-app)).

Reviewer: ```suggestion
* During the [OAuth flow](#installing-the-app), you received and stored the `client_id` and `access_token`.
```

Author: Added the same approach to the line above also 💯

> For context, the [Tax Provider API Connection endpoints](/docs/rest-contracts/tax-app-connection) provide an added layer of security for tax provider connectivity. These basic credentials are used to authenticate each API request to the tax provider from the associated store, allowing you to reject any other unauthorized requests.

Reviewer: ```suggestion
For context, the [Tax Provider API Connection endpoints](/docs/rest-contracts/tax-app-connection) adds a layer of security for tax provider connectivity. These basic credentials are used to authenticate each API request to the tax provider from the associated store, allowing you to reject any other unauthorized requests.
```

> An account login flow allows you to capture basic authentication credentials from merchants. It's best to provide a login flow within your app iFrame so that you can choose your own authentication method. You can also provide an account registration flow in your iFrame to help merchants create a new tax provider account. Learn more about [designing the app UI](/docs/integrations/apps/guide/ui).

Reviewer: ```suggestion
An account login flow lets you capture basic authentication credentials from merchants. We recommend that you provide a login flow within your app iFrame so that you can choose your own authentication method. You can also provide an account registration flow in your iFrame to help merchants create a new tax provider account. Learn more about [designing the app UI](/docs/integrations/apps/guide/ui).
```

> In either case, the [Update Connection](/docs/rest-contracts/tax-app-connection#update-a-connection) endpoint should be called after your app has been successfully installed and the merchant's account credentials have been authenticated. You will need to include `store_hash`, `provider_id`, and `X-Auth-Token`(`access_token`) values.

Reviewer: ```suggestion
We recommend calling the [Update Connection](/docs/rest-contracts/tax-app-connection#update-a-connection) endpoint immediately after the merchant installs your app and you authenticate the merchant's account credentials. Otherwise, your tax provider will not be displayed when merchants navigate to the **Settings > Setup > Tax** page in the control panel. You will need to include `store_hash`, `provider_id`, and `X-Auth-Token`(`access_token`) values.
```

Reviewer: We can remove the next line (repetitive)

Author: Agreed, in the context of establishing a connection it's pretty low value.

We recommend calling the [Update Connection](/docs/rest-contracts/tax-app-connection#update-a-connection) endpoint immediately after the app has been successfully installed, otherwise your tax provider will not be displayed when merchants navigate to the **Settings > Setup > Tax** page in the control panel.

The [Get a Connection](/docs/rest-contracts/tax-app-connection#get-connection-status) request can be used at any time to retrieve the connection status of the specified tax provider in the context of a store.

## Configure tax provider settings in the control panel

> After you have successfully installed your app and provided basic authentication credentials through the [Update Connection](/docs/rest-contracts/tax-app-connection#update-a-connection) request, merchants will be able to adjust their tax provider settings to activate supported functions on their store. 

Reviewer: ```suggestion
After you install your app and provide basic authentication credentials through the [Update Connection](/docs/rest-contracts/tax-app-connection#update-a-connection) request, merchants can adjust their tax provider settings to activate supported functions on their store. 
```

Merchants can adjust these settings by navigating to **Settings > Setup > Tax > _TaxProviderName_** in the control panel. 

Before you can provide tax estimates, the merchant must choose which regions they want you to service. The control panel displays a list of countries and subdivisions that you support. 

If document submission is supported, the merchant can choose to select the **Submit Order Data** checkbox. This will enable document submission on their store, for supported transactions.

## Tax estimation

### When tax estimates are requested

> After the tax provider has been enabled in the control panel, the tax provider is ready to respond to tax estimate requests sent by BigCommerce.

Reviewer: ```suggestion
After the merchant enables your app in the control panel, your app is ready to respond to tax estimate requests sent by BigCommerce.
```

  If you have registered your app in the Developer Portal but have not submitted it for approval because it's still in development, the app will be in a Draft state. This means your app can only be installed on stores owned by the same email address as the Developer Portal account email.
</Callout>

> ## Installing the app

Author: 👁️ I've adjusted the whitespace to remove any double blank lines - they didn't appear to serve a purpose.

Reviewer: 👍

For context, the [Tax Provider API Connection endpoints](/docs/rest-contracts/tax-app-connection) provide an added layer of security for tax providers. They set basic authentication credentials for the tax provider and these basic credentials are used to authenticate each API request to the tax provider from the associated store.

> Offering an account login flow within the tax provider's app iFrame will commonly be required to capture these basic authentication credentials from merchants. An account registration flow may also be helpful for offering merchants the ability to create a new tax provider account. Learn more about designing the app UI [here](/docs/integrations/apps/guide/ui).

Reviewer: ```suggestion
An account login flow allows you to capture basic authentication credentials from merchants. It is best to provide a login flow within your app iFrame so that you can choose your own authentication method. You can also provide an account registration flow in your iFrame to help merchants create a new tax provider account. Learn more about [designing the app UI](/docs/integrations/apps/guide/ui).
```

We recommend calling the [Update Connection](/docs/rest-contracts/tax-app-connection#update-a-connection) endpoint immediately after the app has been successfully installed, otherwise your tax provider will not be displayed when merchants navigate to the **Settings > Setup > Tax** page in the control panel.

The [Get a Connection](/docs/rest-contracts/tax-app-connection#get-connection-status) request may be used at any time to retrieve the connection status of the specified tax provider in the context of a store.

## Configure tax provider settings in the control panel

Once you have successfully installed the tax provider's app and provided basic authentication credentials through the [Update Connection](/docs/rest-contracts/tax-app-connection#update-a-connection) request, merchants will be able to adjust the tax provider's settings to activate supported functions on their store. 

> The tax provider's settings screen can be found by navigating to **Settings > Setup > Tax > _TaxProviderName_** in the control panel. 

Reviewer: ```suggestion
Merchants can adjust these settings by navigating to **Settings > Setup > Tax > _TaxProviderName_** in the control panel. 
```

> The tax provider's settings screen can be found by navigating to **Settings > Setup > Tax > _TaxProviderName_** in the control panel. 

Reviewer: will add to the linter dictionary

> To enable the tax provider to respond to tax estimate requests, the merchant must choose which supported countries or subdivisions the tax provider will fulfill tax quotes for.

Reviewer: ```suggestion
Before you can provide tax estimates, the merchant must choose which tax zones they want you to service. The control panel displays a list of countries and subdivisions that you support. 
```

For context, the [Tax Provider API Connection endpoints](/docs/rest-contracts/tax-app-connection) provide an added layer of security for tax providers. They set basic authentication credentials for the tax provider and these basic credentials are used to authenticate each API request to the tax provider from the associated store.

> Offering an account login flow within the tax provider's app iFrame will commonly be required to capture these basic authentication credentials from merchants. An account registration flow may also be helpful for offering merchants the ability to create a new tax provider account. Learn more about designing the app UI [here](/docs/integrations/apps/guide/ui).

Author: Looking to speak distinctly to the `login flow` vs the `registration flow`.

Reviewer: Thanks, I added a suggestion [below](https://github.com/bigcommerce/docs/pull/130#discussion_r1516764263)

> In either case, the [Update Connection](/docs/rest-contracts/tax-app-connection#update-a-connection) endpoint should be called after the tax provider's app has been successfully installed and the merchant's account credentials have been authenticated. Tax providers will need to include `store_hash`, `provider_id`, and `X-Auth-Token`(`access_token`) values.

Author: We've repeatedly seen scenarios where a tax provider has reached out to support, where the app is installed but a connection has bot been set up. Hopefully this helps guide this area a little further.

The [Get a Connection](/docs/rest-contracts/tax-app-connection#get-connection-status) request may be used at any time to retrieve the connection status of the specified tax provider in the context of a store.

## Configure tax provider settings in the control panel

Once you have successfully installed the tax provider's app and provided basic authentication credentials through the [Update Connection](/docs/rest-contracts/tax-app-connection#update-a-connection) request, merchants will be able to adjust the tax provider's settings to activate supported functions on their store. 

The tax provider's settings screen can be found by navigating to **Settings > Setup > Tax > _TaxProviderName_** in the control panel. 

To enable the tax provider to respond to tax estimate requests, the merchant must choose which supported countries or subdivisions the tax provider will fulfill tax quotes for.

> If document submission is supported by the tax provider, the merchant can choose to select the **Submit Order Data** checkbox. This will enable document submission on their store, for supported transactions.

Author: In the above section we are introducing a significant change to the process in that we need the merchant to select supported countries or subdivisions in order for tax estimation to function.

e.g. A tax provider may support Australia and New Zealand, and the merchant may choose to select that only Australian based transactions are handled by the tax provider, for their purpose.

Reviewer: 👍

Estimate requests are not expected after the following events.

> * While browsing a store's home page, product catalog or product pages.

Author: Clarifying that the home page is also not ever an expected place that a estimate request will occur (on the back of a ticket that identified an edge case where it was).

Reviewer: 👍

  Tax estimate requests sent by BigCommerce may not always contain complete data as these requests will be fired at different stages of the shopper checkout. For example, the **Estimate Shipping & Tax** functionality on the cart page does not provide any billing address data, but the tax provider will still be expected to return a valid tax estimate.
</Callout>

### Responding to tax estimate requests

> When responding to tax estimate requests sent by BigCommerce, you are required to include aggregates and breakdowns of sales tax amounts and rates for product item prices, shipping and handling prices. This is because when navigating to the **Settings > Setup > Tax > Tax Settings** page in the control panel, merchants have the ability to specify whether they would like to show taxes in cart, checkout, orders and invoice **As one summarized line item** or **Broken down by tax rate**.

Reviewer: fixed in suggestion

[View the Estimate Taxes API reference](/docs/rest-contracts/tax#estimate-taxes).

## Document submission

Document submission enables you to persist a tax quote request, replace it with another one, or invalidate it if necessary.

> Supporting document submission is optional. However, if you wish to support this functionality you must share the required document submission URLs with BigCommerce when [sharing provider details with BigCommerce](#sharing-provider-details-with-bigcommerce).

Reviewer: fixed in suggestion

> Supporting document submission is optional. However, if you wish to support this functionality you must share the required document submission URLs with BigCommerce when [sharing provider details with BigCommerce](#sharing-provider-details-with-bigcommerce).

Reviewer: fixed in suggestion

> If your tax provider supports document submission, merchants can navigate to **Settings > Setup > Tax > _TaxProviderName_** in the control panel and enable the **Submit Order Data** checkbox to active document submission operations.

Reviewer: fixed in suggestion

[View the Commit Tax Quote API reference](/docs/rest-contracts/tax#commit-tax-quote).

## Testing

> Prior to testing your tax provider, your store should have the following configured in the control panel:

Reviewer: fixed in suggestion

> Prior to testing your tax provider, your store should have the following configured in the control panel:

Reviewer: fixed in suggestion

* The shipping origin address, found by navigating to **Settings > Setup > Shipping**, is configured. This value is included in tax estimate requests
* The tax provider, found by navigating to **Settings > Setup > Tax > _TaxProviderName_**, is enabled in the relevant countries or subdivisions
* If document submission is supported, navigate to **Settings > Setup > Tax > _TaxProviderName_** and ensure the **Submit Order Data** checkbox is checked

> To test the tax provider connection, navigate to **Settings > Setup > Tax > _TaxProviderName_** in the control panel and click **Test Connection**. This function will trigger a sample tax estimate request to the tax provider's estimate URL. If the connection is unsuccessful, or if there is an issue with the response, merchants can navigate to **Settings > Advanced > Store logs** in the control panel to view the error and its description to assist with triaging the issue.

Reviewer: fixed in suggestion

> To test the tax provider connection, navigate to **Settings > Setup > Tax > _TaxProviderName_** in the control panel and click **Test Connection**. This function will trigger a sample tax estimate request to the tax provider's estimate URL. If the connection is unsuccessful, or if there is an issue with the response, merchants can navigate to **Settings > Advanced > Store logs** in the control panel to view the error and its description to assist with triaging the issue.

Reviewer: fixed in suggestion

  Tax estimate requests sent by BigCommerce may not always contain complete data as these requests will be fired at different stages of the shopper checkout. For example, the **Estimate Shipping & Tax** functionality on the cart page does not provide any billing address data, but the tax provider will still be expected to return a valid tax estimate.
</Callout>

### Responding to tax estimate requests

> When responding to tax estimate requests sent by BigCommerce, you are required to include aggregates and breakdowns of sales tax amounts and rates for product item prices, shipping and handling prices. This is because when navigating to the **Settings > Setup > Tax > Tax Settings** page in the control panel, merchants have the ability to specify whether they would like to show taxes in cart, checkout, orders and invoice **As one summarized line item** or **Broken down by tax rate**.

Reviewer: ```suggestion
When you respond to BigCommerce's requests for tax estimates, you must include aggregates and breakdowns of sales tax amounts and rates for product item prices, shipping and handling prices. This is because merchants can specify whether they want to show taxes in cart, checkout, orders and invoice **As one summarized line item** or **Broken down by tax rate**.
```

^ How merchants navigate the control panel seems irrelevant here (and the linter is telling me the sentence is hard to read)

Author: Good call, thanks @bc-andreadao!

[View the Estimate Taxes API reference](/docs/rest-contracts/tax#estimate-taxes).

## Document submission

Document submission enables you to persist a tax quote request, replace it with another one, or invalidate it if necessary.

> Supporting document submission is optional. However, if you wish to support this functionality you must share the required document submission URLs with BigCommerce when [sharing provider details with BigCommerce](#sharing-provider-details-with-bigcommerce).

Reviewer: ```suggestion
Supporting document submission is optional. If you want to support this functionality, you must share document submission URLs with BigCommerce when [sharing provider details](#sharing-provider-details-with-bigcommerce).
```

> If your tax provider supports document submission, merchants can navigate to **Settings > Setup > Tax > _TaxProviderName_** in the control panel and enable the **Submit Order Data** checkbox to active document submission operations.

Reviewer: ```suggestion
If your tax provider supports document submission, merchants can enable this feature by navigating to **Settings > Setup > Tax > _TaxProviderName_** in the control panel and checking the **Submit Order Data** checkbox.
```

[View the Commit Tax Quote API reference](/docs/rest-contracts/tax#commit-tax-quote).

## Testing

> Prior to testing your tax provider, your store should have the following configured in the control panel:

Reviewer: ```suggestion
Before testing your tax provider, your store should have the following configured in the control panel:
```

* The shipping origin address, found by navigating to **Settings > Setup > Shipping**, is configured. This value is included in tax estimate requests
* The tax provider, found by navigating to **Settings > Setup > Tax > _TaxProviderName_**, is enabled in the relevant countries or subdivisions
* If document submission is supported, navigate to **Settings > Setup > Tax > _TaxProviderName_** and ensure the **Submit Order Data** checkbox is checked

> To test the tax provider connection, navigate to **Settings > Setup > Tax > _TaxProviderName_** in the control panel and click **Test Connection**. This function will trigger a sample tax estimate request to the tax provider's estimate URL. If the connection is unsuccessful, or if there is an issue with the response, merchants can navigate to **Settings > Advanced > Store logs** in the control panel to view the error and its description to assist with triaging the issue.

Reviewer: ```suggestion
To test the tax provider connection, navigate to **Settings > Setup > Tax > _TaxProviderName_** in the control panel and click **Test Connection**. This function will trigger a request to the tax provider's estimate URL for a sample tax estimate. If the connection is unsuccessful, or if there is an issue with the response, merchants can navigate to **Settings > Advanced > Store logs** in the control panel to view the error and its description to help triage the issue.
```

Prior to testing a tax provider, the merchant or partner test store should have the following configured in the control panel:

* The shipping origin address, found by navigating to **Settings > Setup > Shipping**, is configured. This value is included in tax estimate requests

> * The tax provider, found by navigating to **Settings > Setup > Tax > _TaxProviderName_**, is enabled in the relevant countries or subdivisions

Reviewer: will add to dictionary

Prior to testing a tax provider, the merchant or partner test store should have the following configured in the control panel:

> * The store default country, found by navigating to **Settings > Setup > Store profile**, is set to a country that is supported by the tax provider

Author: We are removing this requirement from the flow now.

That is, even if the store is based in Australia the store will be able to see TaxJar (United States only tax provider) and select it to be enabled for United States transactions.

That is, we don't want to get in the way of offering flexible international selling configuration.

* The tax provider, found by navigating to **Settings > Setup > Tax > _TaxProviderName_**, is enabled in the relevant countries or subdivisions
* If document submission is supported, navigate to **Settings > Setup > Tax > _TaxProviderName_** and ensure the **Submit Order Data** checkbox is checked

> To test the tax provider connection, navigate to **Settings > Setup > Tax > _TaxProviderName_** in the control panel and click **Test Connection**. This selection will trigger a sample tax estimate request to be sent to the estimate URL provided by the tax provider. If the connection is unsuccessful, or there is an issue with the response, merchants can navigate to **Settings > Advanced > Store logs** in the control panel to view the error and its description to assist with triaging the issue.

Author: `Test connection` really validates against a complex tax quote, so we've added some more detail here to explain that the response may also be awry (not just the connection itself).

Reviewer: 👍
